src/Profiling.cpp

```diff
@@ -206,8 +211,27 @@ class InjectProfiling : public IRMutator {
 
     void visit(const For *op) {
         // We profile by storing a token to global memory, so don't enter GPU loops
-        if (op->device_api == DeviceAPI::None ||
-            op->device_api == DeviceAPI::Host) {
+        if (op->device_api == DeviceAPI::Hexagon) {
```

Comment:
While I don't have a strong objection to making this Hexagon specific, this looks like a general "RPC like target" feature. I expect we'll be seeing more use of the RPC idea, e.g. for remote execution across clusters. Thus we should probably be looking forward on such. One possibility is to introduce some abstraction of a "remote device" type of thing. Other possibility is to put in a TODO comment with some tag like "remote device support" so we can keep track of places where general functionality should go.